docs/release.md

```diff
@@ -0,0 +1,180 @@
+
+Run prowgen to validate configuration:
+```
+docker pull registry.svc.ci.openshift.org/ci/ci-operator-prowgen:latest
+docker run -it -v $(pwd)/ci-operator:/ci-operator:z registry.svc.ci.openshift.org/ci/ci-operator-prowgen:latest --from-dir /ci-operator/config/ --to-dir /ci-operator/jobs
+```
+
+Merge it to release repository. This will make sure that any PR to release branches is gates by tests.
+
+3. Create release PR into `openshift/openshift-azure` repository release branch:
+
+```
+git checkout upstream/release-vx
+git checkout -b release-vx-pluginconfig
+```
+
+update `pluginconfig/pluginconfig-311.yaml` file with image version for this particular release
+```
+clusterVersion: vx.y
+imageVersion: 311.69.20190214
+images:
+  ansibleServiceBroker: registry.access.redhat.com/openshift3/ose-ansible-service-broker:v3.11.69
+  azureControllers: quay.io/openshift-on-azure/azure-controllers:vx.y
+  etcdBackup: quay.io/openshift-on-azure/etcdbackup:vx.y
+```
+
+Make sure you update `clusterVersion` field all image version to point to specific version, instead of `latest` tag.
+
+Merge this PR into `release-vx` branch. Test should be using test infrastructure built images.
+
+4. Tag release for the release you just merged in step 3.
+
+This step requires elevated right on git repo. You might need to ask somebody else to execute these commands.
+
+```
+git checkout upstream/release-vx
+git tag -a vx.y -m 'reason' # where reason is release summary in one sentence
+git push upstream tags/vx.y
+```
+
+1. Build release images from release tags
+
+This step should be executed only when PR from step 3 is merged and 4 is pushed.
+
+```
+git checkout tags/vx.y
+# validate script can see right tag
+make version
+<you should see tag version as an output. It will be used to tag and publish images>
+make azure-controllers-push metricsbridge-push sync-push etcdbackup-push
+```
+
+6. Update upgrade tests for published release in `openshift/release`
+
+This step can be done together with step 2, but for first release you are doing we recommend to do it as separate PR.
+
+```
+git checkout -b osa.release.vx.testing
+```
+
+Add vx.y target to `ci-operator/config/openshift/openshift-azure/openshift-openshift-azure-master.yaml`
+
+```
+- artifact_dir: /tmp/artifacts
+ as: e2e-upgrade-vx.y
+ commands: ARTIFACT_DIR=/tmp/artifacts SOURCE=vx.y make upgrade
+ secret:
+   name: azure
+   mount_path: /usr/secrets
+ container:
+   from: src
+```
+
+Run prowgen to generate test jobs:
+```
+docker pull registry.svc.ci.openshift.org/ci/ci-operator-prowgen:latest 
+docker run -it -v $(pwd)/ci-operator:/ci-operator:z registry.svc.ci.openshift.org/ci/ci-operator-prowgen:latest --from-dir /ci-operator/config/ --to-dir /ci-operator/jobs
+```
+
+Change generated jobs to `always_run: false` and `optional: true`
+We not gonna use automatic generated jobs with `e2e-upgrade-vx.y` because of lack of targets.
```

Comment:
consider: s/We not gonna/We are not going to/

Reply:
I don't understand what this line means and I think I need to.